docs: add `compile-html` (#1751)
commercetools/merchant-center-application-kit: website/src/content/deployment/compile-html.mdx

---

> title: Compiling Custom Application

Reviewer: Should we name this file `compiling-index-html.mdx`? And the title `Compiling index.html`?

Just looking at this from the navigation menu links perspective...

Reviewer: Technically it's only about compiling the `index.html` file. Maybe something like `Compiling the application for static hosting`, or `Preparing the application for static hosting`?

We can also get some input from others (cc @tdeekens @anilkk )

PS: The file name should more or less match the title.

Reviewer 2: I don't have strong opinions here apart from title matching filename. Given we ever compile more html than the `index.html` keeping it "generic" could help. As a consumer I also don't care that the thing I compile is an `index.html` it's more about the fact that I have to. Maybe "Compile Custom Application index file"

---

> One of the necessary steps to prepare your Custom Application for a deployment is to create a [production build](/deployment/production-build). If you decide to host you Custom Application statically and not use `mc-http-server`, you need to compile your Custom Application.

Reviewer 2: ```suggestion
One of the necessary steps to prepare your Custom Application for a deployment is to create a [production build](/deployment/production-build). If you decide to host your Custom Application statically and do not intend to use the `mc-http-server`, you need to compile your Custom Application.
```

This section helps prepare your Custom Application for deployment by compiling it.

# Why compile?

When creating a [production build](/deployment/production-build), `mc-scripts build` takes your Custom Application code and generates assets that can be served statically.

> One of these assets is a file `index.html.template`. This file contains references to bundles (your code and other assets) and placeholders that needs to be replaced with values of your [Custom Application configuration](/development/application-config) as runtime application values. In this step is where compiling is needed.

Reviewer 2: ```suggestion
One of these assets is a file `index.html.template`. This file contains references to bundles (your code and other assets) and placeholders that needs to be replaced with values specific to your [Custom Application configuration](/development/application-config). In this step compilation is needed.
```

Once your Custom Application is compiled, it is ready for deployment.

# Prerequisites

Before you get started, you need to do the following:

- [Configure your Custom Application](/development/application-config).
- Create a [production build](/deployment/production-build) of your Custom Application.

# Compiling your Custom Application

> To compile your Custom Application, you run the following:

Reviewer 2: ```suggestion
Run the following to compile your Custom Application:
```

```console noPromptLines="2,4-6"
yarn compile-html --use-local-assets
```

<Info>

> The `--use-local-assets` option is required for the sake of this example. We also strongly recommend that you host static assets to a CDN or Cloud Storage. See [Serving static assets](/deployment/serving-static-assets).

Reviewer 2: ```suggestion
The `--use-local-assets` option is only needed in this example. We strongly recommend that you do not use local assets but serve static assets from a CDN or Cloud Storage. See [Serving static assets](/deployment/serving-static-assets).
```

> The `--use-local-assets` option is only needed in this example. We strongly recommend that you do not use local assets but serve static assets from a CDN or Cloud Storage. See [Serving static assets](/deployment/serving-static-assets).

Author: I'm not really sure what you mean.

```
- `--use-local-assets`: Use this option to include references of your local assets when compiling
```

under this section?
https://github.com/commercetools/merchant-center-application-kit/blob/master/website/src/content/deployment/serving-static-assets.mdx#serving-assets-on-a-remote-cdn-or-cloud-storage

> The `--use-local-assets` option is only needed in this example. We strongly recommend that you do not use local assets but serve static assets from a CDN or Cloud Storage. See [Serving static assets](/deployment/serving-static-assets).

Reviewer: We have this for the HTTP server: https://docs.commercetools.com/custom-applications/deployment/serving-static-assets#serving-assets-using-the-http-server

We should also have one for the static hosting as well (for consistency).

> The `--use-local-assets` option is only needed in this example. We strongly recommend that you do not use local assets but serve static assets from a CDN or Cloud Storage. See [Serving static assets](/deployment/serving-static-assets).

Author: if you referring to having a bullet list, then we are saying the same thing 😉

</Info>

> The command above does what we need: it compiles the `index.html` using the JavaScript bundle references (after running `mc-scripts build`) and the runtime configuration. At this point the `index.html` file is ready for production usage.

Reviewer 2: ```suggestion
The command above does what we need: it compiles a `index.html` using the references to the JavaScript bundle (created by running `mc-scripts build`). It also replaces the needed runtime configuration. At this point the `index.html` file is ready for production usage.
```
